src/utils/MasterSlave.cpp

```diff
@@ -161,16 +161,23 @@ void MasterSlave::reduceSum(double *sendData, double *rcvData, int size)
 
   if (_isSlave) {
     // send local result to master
-    _communication->reduceSum(sendData, rcvData, size, 0);
+    _communication->reduceSum(sendData, rcvData, 0);
   }
 
   if (_isMaster) {
     // receive local results from slaves, apply SUM
-    _communication->reduceSum(sendData, rcvData, size);
+    _communication->reduceSum(sendData, rcvData);
   }
 }
 
-void MasterSlave::reduceSum(int &sendData, int &rcvData, int size)
+void MasterSlave::reduceSum(const double &sendData, double &rcvData)
+{
+  PRECICE_TRACE();
+  reduceSum(precice::span<const double>{&sendData, 1},
+            precice::span<double>{&rcvData, 1});
+}
```

Comment:
Why is `void MasterSlave::reduceSum(const double &sendData, double &rcvData)` so short and `void MasterSlave::reduceSum(const int &sendData, int &rcvData)` so long? Shouldn't both functions look (more or less) the same?

Reply:
I reused the reduceSum for "multiple doubles" to implement "one double".
There is no "multiple ints" version, so I had to keep that.